src/lib/block/kuznyechik/kuznyechik.cpp

```diff
@@ -4257,43 +4237,59 @@ bool Kuznyechik::has_keying_material() const {
 }
 
 void Kuznyechik::key_schedule(std::span<const uint8_t> key) {
+   using namespace Kuznyechik_F;
+
    BOTAN_ASSERT_NOMSG(key.size() == 32);
 
-   uint64_t k0, k1, k2, k3;
-   k0 = load_le<uint64_t>(key.data(), 0);
-   k1 = load_le<uint64_t>(key.data(), 1);
-   k2 = load_le<uint64_t>(key.data(), 2);
-   k3 = load_le<uint64_t>(key.data(), 3);
+   uint64_t k0 = load_le<uint64_t>(key.data(), 0);
+   uint64_t k1 = load_le<uint64_t>(key.data(), 1);
+   uint64_t k2 = load_le<uint64_t>(key.data(), 2);
+   uint64_t k3 = load_le<uint64_t>(key.data(), 3);
```

Comment:
FYI: With #3707 I so far didn't add overloads for `load_le(std::span<>, size_t offset);`. My assumption was that we'd either use `BufferSlicer` to do the offsetting explicitly, or formulate such stride-loads as:

```C++
uint64_t k0, k1, k2, k3;
load_le(key, k0, k1, k2, k3);
```

To avoid pre-defining variables and initializing them later, a (not yet implemented) option might be:

```C++
auto [k0, k1, k2, k3] = load_le<uint64_t>(key, 4 /* number of uint64_t to load */);
```

Might be worth discussing this design in #3707.

Reply:
The destructuring approach would be nice. I generally don't like leaving unitialized vars but in the sort of code we'd be looking at (like your first snippet) they are immediately initialized in the next line so ... meh. It would be fairly easy to typo and leave something uninitialized (eg `load_le(key, k0, k1, k2, k2)`), unlike the approach that returns a tuple, but that situation should be caught by any reasonable tests.